sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -583,6 +583,16 @@ case class Aggregate(
   }
 }
 
+case class AggregateWithHaving(
```

Comment:
Could we rename this into `UnresolvedHaving`, then move it into `unresolved.scala`? I personally think `HAVING` always comes with `Aggregate`, so we don't need to include `Aggregate` in the name.

Reply:
```
move it into unresolved.scala?
```
Yeah, make sense, will change it to unresolved.scala.

```
Could we rename this into UnresolvedHaving
```
Since the `group by` not always come with `Aggregate`, it can also be `GroupingSets`,  we only handle the Aggregate part with `AggregateWithHaving`. So maybe let's keep it `AggregateWithHaving`? WDYT :)